src/Components/Components/src/Routing/Router.cs

```diff
@@ -190,28 +192,31 @@ private void Refresh(bool isNavigationIntercepted)
             }
         }
 
-        private async Task RunOnNavigateAsync(string path)
+        private async ValueTask<bool> RunOnNavigateAsync(string path, Task prevOnNavigate)
         {
             // If this router instance does not provide an OnNavigateAsync parameter
             // then we render the component associated with the route as per usual.
             if (!OnNavigateAsync.HasDelegate)
             {
-                return;
+                return true;
```

Comment:
If I follow the code correctly, RefreshRouteTable is invoked in this case. Why would we need to do that?

Reply:
If this returns `true`, then we call `Refresh` which renders the route's component. `RefreshRouteTable` is called inside `Refresh` but will only change the route table if `AdditionalAssemblies` is modified.